code/modules/clothing/shoes/miscellaneous.dm

```diff
@@ -213,16 +213,17 @@
 		return
 
 	var/atom/target = get_edge_target_turf(user, user.dir) //gets the user's direction
-
-	if (user.throw_at(target, jumpdistance, jumpspeed, spin = FALSE, diagonals_first = TRUE, callback = CALLBACK(src, .proc/hop_end)))
+	if (user.throw_at(target, jumpdistance, jumpspeed, spin = FALSE, diagonals_first = TRUE, callback = CALLBACK(src, .proc/hop_end, list(user))))
+		user.movement_type |= FLYING
 		jumping = TRUE
 		playsound(src, 'sound/effects/stealthoff.ogg', 50, 1, 1)
 		user.visible_message("<span class='warning'>[usr] dashes forward into the air!</span>")
 	else
 		to_chat(user, "<span class='warning'>Something prevents you from dashing forward!</span>")
 
-/obj/item/clothing/shoes/bhop/proc/hop_end()
+/obj/item/clothing/shoes/bhop/proc/hop_end(mob/user)
 	jumping = FALSE
+	user.movement_type &= ~FLYING
```

Comment:
needs a way to prevent removing flying from mobs already having the flag when they jumped(you already know this but just so no one merges/misses it)